docs/tutorials/Remark.adoc

```diff
@@ -0,0 +1,396 @@
+
+== Updating Storage
+
+AddressBook stores data by serializing `JsonAdaptedPerson` into `json` with the help of an external library -- Jackson.
+Let's update `JsonAdaptedPerson` to work with our new `Person`!
+
+While the changes to code may be minimal, the test data will have to be updated as well.
+
+Check out link:https://github.com/j-lum/addressbook-level35/commit/78b2a7438a9f9d828b7f09e190ff6130790966c0[this commit] to see what the changes entail.
+
+== Finalizing the UI
+
+Now that we have finalized the `Person` class and its dependencies, we can now bind the `Remark` field to the UI.
+
+Just add link:https://github.com/j-lum/addressbook-level35/commit/2edaf1c52e07e28dc7a49742bc63352813a55af9[this one line of code!]
+[source, java]
+.PersonCard.java
+----
+public PersonCard(Person person, int displayedIndex) {
+    //...
+    remark.setText(person.getRemark().value);
+}
+----
+
+.The remark label is bound properly!
+image::RemarkBound.png[]
+
+== Putting everything together
+
+After the previous step, we notice a peculiar regression -- we went from displaying something to nothing at all.
+However, this is expected behavior as we have never changed `RemarkCommand` at all!
+
+=== Update `RemarkCommand`
+
+In this last step, we modify `RemarkCommand#execute()` to change the `Remark` of a `Person`.
+Since all fields in a `Person` are immutable, we create a new instance of a `Person` with the values that we want and save it with `Model#setPerson()`.
+
+[source, java]
+.RemarkCommand.java
+----
+//...
+    public static final String MESSAGE_ADD_REMARK_SUCCESS = "Added remark to Person: %1$s";
+    public static final String MESSAGE_DELETE_REMARK_SUCCESS = "Removed remark from Person: %1$s";
+//...
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(index.getZeroBased());
+        Person editedPerson = new Person(personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
+                personToEdit.getAddress(), remark, personToEdit.getTags());
+
+        model.setPerson(personToEdit, editedPerson);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+        return new CommandResult(generateSuccessMessage(editedPerson));
+    }
+
+    /**
+     * Generates a command execution success message based on whether the remark is added to or removed from
+     * {@code personToEdit}.
+     */
+    private String generateSuccessMessage(Person personToEdit) {
+        String message = !remark.value.isEmpty() ? MESSAGE_ADD_REMARK_SUCCESS : MESSAGE_DELETE_REMARK_SUCCESS;
+        return String.format(message, personToEdit);
+    }
+----
```

Comment:
Code block also does not show to implement `equals`
See: https://github.com/j-lum/addressbook-level3/blob/1492fb0d4e8a075f2481028377e0d113e7c01b76/src/main/java/seedu/address/logic/commands/RemarkCommand.java

Reply:
Added the implementation in a previous step.